src/seth/libexec/seth/seth-calldata

```diff
@@ -1,43 +1,36 @@
-#!/usr/bin/env node
-const usage = `Usage: seth calldata <sig> [<args>]>
-   or: seth calldata <file>
-   or: seth calldata <data>`;
-if (process.argv.length >= 3 &&
-    process.argv[2].indexOf('(') >= 0) {
-  // silence warnings we don't care about
-  const log = console.log
-  console.log = () => {};
-  const ethers = require("./ethers.min.js");
-  console.log = log;
-  const sig = process.argv[2];
-  try {
-    const funcs = new ethers.utils.Interface(["function " + sig.replace(')(', ') returns (')]).functions;
-    const args = process.argv.slice(3).map(s => {
-      try {
-        return JSON.parse(s)
-      }
-      catch (e) {
-        return s
-      }
-    })
-    console.log(funcs[Object.keys(funcs)[0]].encode(args))
-  } catch (e) {
-    console.error(e.toString())
-    process.exit(1)
-  }
-} else if (process.argv.length == 3) {
-  const {execFileSync}   = require('child_process');
-  try {
-    const yes = execFileSync("seth",
-                             ["--to-hexdata",
-                              process.argv[2]]
-                            ).toString().replace('\n','')
-    console.log(yes)
-  } catch(e) {
-    console.error(e)
-    process.exit(1)
-  }
-} else {
-  console.error(usage)
-  process.exit(1)
-}
+#!/usr/bin/env bash
+### seth-calldata -- pack a signature and a list of arguments into calldata
+### Usage: seth calldata <sig> [<args>]
+###    or: seth calldata <file>
+###    or: seth calldata <data>
+###
+### Pack <sig> and <args> into data suitable for passing to `seth call'.
+### If <sig> has the form `<name>(<types>)', pack <sig>/<args> into calldata.
+###
+### If <file> contains a slash character, treat it as a file name to read.
+### Otherwise, ensure <data> begins with `0x' and treat it as hexdata.
+set -e
+if [[ $1 =~ ^([^\(]+)\( ]]; then
+  name=${BASH_REMATCH[1]}
+  abi=$(seth --abi-function-json "$1" | jq .[0])
+  shift
+  for arg; do
```

Comment:
What is `arg` here?

Reply:
the arguments given to `seth calldata` after the signature. Don't really know how this works, it's mainly adapted from the [previous version of `seth calldata` ](https://github.com/dapphub/dapptools/blob/78508c6a8db2d6d3e8e09437dbe122bb5e6b2e7e/src/seth/libexec/seth/seth-calldata)

Reply:
OK this is [apparently](https://www.gnu.org/software/bash/manual/html_node/Looping-Constructs.html) a horrible Bash special-caseism.

I would rather be explicit here and have `in "$@"`.